### PR TITLE
Implemented customizable translation formatting

### LINKED ISSUE #720: HEX/RGB color codes in "com\velocitypowered\proxy\l10n"
I tried everything to call RGB colors but failed so I think other than basic minecraft colors are not supported.

So i want to make a suggestion because of the RGB colors.

My whole server would be made up of RGB colors, and it would be embarrassing if my proxy server weren't exactly that.
(velocity.jar\com\velocitypowered\proxy\l10n)

### PR DESCRIPTION
**Pull request by 4drian3d**
Added ability to customize the translation formatting

img

resolves #720

### CODE REVIEW

File: proxy/src/main/java/com/velocitypowered/proxy/util/ClosestLocaleMatcher.java [hunk 1]
```diff
@@ -54,4 +61,43 @@ public void registerKnown(final Locale locale) {
...
+        ))
+        .format(format(arguments), new StringBuffer(), null).toString())
+      .colorIfAbsent(color);
+  }
+
+  private static final Object[] format(Object... arguments) {
+    if (arguments == null || arguments.length == 0) {
+      return arguments;
+    }
+    for (int i = 0; i <= 0; i++) {
```
Comment (Joo200) [inline proxy/src/main/java/com/velocitypowered/proxy/util/ClosestLocaleMatcher.java]: This does only work when you're providing exactly one argument.
Your ArrayIndexOutOfBoundException is related to that: The correct iteration for an array is `for (int i = 0; i < array.length; i++)`

Comment (4drian3d) [reply proxy/src/main/java/com/velocitypowered/proxy/util/ClosestLocaleMatcher.java]: Thanks for commenting, I really didn't notice that mistake, thank you very much

Comment (Joo200) [reply proxy/src/main/java/com/velocitypowered/proxy/util/ClosestLocaleMatcher.java]: Now you'll receive every time a ArrayIndexOutOfBoundException...

Comment (4drian3d) [reply proxy/src/main/java/com/velocitypowered/proxy/util/ClosestLocaleMatcher.java]: I made the change without seeing it much, thanks again

**Comment by 4drian3d** [comment]
superseded by #1108

**Pull request closed without merging**